specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2019-05-01-preview/appplatform.json

```diff
@@ -2043,6 +2043,10 @@
           "$ref": "#/definitions/AppResourceProperties",
           "description": "Properties of the App resource",
           "x-ms-client-flatten": false
+        },
+        "location": {
```

Comment:
>location [](start = 9, length = 8)

Please mark as read-only if it's inherited from parent resource.

Reply:
Done

Reply:
If user put a location different from parent resource deliberately, a 404 error will be returned due to the uniqueness of parent resource

Reply:
Your create and update examples don't have location in request body. If it's required then it should be marked required in the swagger and in those examples.

The experience around this is kind of weird. We know what the only valid location is and we're going to make the user put it in the request anyway, but if they get it wrong we will reject the request. Why do we make them specify the location at all?

---
In reply to: [401348345](https://github.com/Azure/azure-rest-api-specs/pull/8851#discussion_r401348345) [](ancestors = 401348345)

Reply:
Updated the create and update examples. Actually the location is required only when enabling/disabling MSI via PUT/PATCH, so ARM can figure out which regional MSI RP it should call. Otherwise, it's optional.